code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -80,16 +80,85 @@
 	desc = "A sturdy shotgun with a longer magazine and a fixed tactical stock designed for non-lethal riot control."
 	icon_state = "riotshotgun"
 	mag_type = /obj/item/ammo_box/magazine/internal/shot/riot
-	sawn_desc = "Come with me if you want to live."
+	var/style = 1
 
 /obj/item/gun/projectile/shotgun/riot/attackby(obj/item/A, mob/user, params)
 	..()
 	if(istype(A, /obj/item/circular_saw) || istype(A, /obj/item/gun/energy/plasmacutter))
-		sawoff(user)
+		sawoff(A, user, 0)
 	if(istype(A, /obj/item/melee/energy))
 		var/obj/item/melee/energy/W = A
 		if(W.active)
-			sawoff(user)
+			sawoff(A, user, 0)
+	if(istype(A, /obj/item/pipe))
+		sawoff(A, user, 1)
+
+/obj/item/gun/projectile/shotgun/riot/sawoff(obj/item/A, mob/user, style = 0)

```

Comment:
Rather than override this, you should do the following:
1. At code/modules/projectiles/guns/projectile.dm, cut from line 169 to 177 and put them in a new proc, as such
```
/obj/item/gun/projectile/proc/post_sawoff(mob/user)
    user.visible_message("[user] shortens \the [src]!", "<span class='notice'>You shorten \the [src].</span>")
    name = "sawn-off [name]"
    desc = sawn_desc
    w_class = WEIGHT_CLASS_NORMAL
    item_state = "gun"//phil235 is it different with different skin?
    slot_flags &= ~SLOT_BACK    //you can't sling it on your back
    slot_flags |= SLOT_BELT     //but you can wear it on your belt (poorly concealed under a trenchcoat, ideally)
    sawn_state = SAWN_OFF
    update_icon()
```
then put `post_sawoff(user)` on line 169, between `<tab>return` and `return 1`.
 2. Add a new proc at line 179 (basically right underneath the sawoff proc) called `unsaw(mob/user)`. Populate it with the following:
```
/obj/item/gun/projectile/proc/unsaw(mob/user)
    if(sawn_state == SAWN_INTACT)
        to_chat(user, "<span class='warning'>[src] has not been shortened!</span>")
        return
    user.changeNext_move(CLICK_CD_MELEE)
    user.visible_message("<span class='notice'>[user] begins to lengthen [src].</span>", "<span class='notice'>You begin to lengthen [src]...</span>")

    if(blow_up(user))
        user.visible_message("<span class='danger'>[src] goes off!</span>", "<span class='danger'>[src] goes off in your face!</span>")
        return

    if(do_after(user, 30, target = src))
        if(sawn_state == SAWN_INTACT)
            return
        post_unsaw(user)
        return 1
```
 3. Add a new proc, `post_unsaw(mob/user)`
```
/obj/item/gun/projectile/proc/post_unsaw(mob/user)
    user.visible_message("<span class='notice'>[user] lengthens [src]!</span>", "<span class='notice'>You lengthen [src].</span>")
    name = initial(name)
    desc = initial(desc)
    w_class = initial(w_class)
    item_state = initial(item_state)
    slot_flags &= ~SLOT_BELT
    slot_flags |= SLOT_BACK
    sawn_state = SAWN_INTACT
```
 4. Override post_sawoff and post_unsaw for the riot shotgun to do anything extra you'd like, using `..()` at the top to run all of the parent behaviors. This probably means changing current_skin and magazine.max_ammo, I'll let you figure out how to do this so I don't spoil all the fun.